src/amp-url-creator.js

```diff
@@ -35,6 +46,55 @@ const HAS_RTL_CHARS = new RegExp('[' + RTL_CHARS + ']');
 /** @private {number} */
 const MAX_DOMAIN_LABEL_LENGTH_ = 63;
 
+
+/**
+ * Constructs a Viewer proxy url using these rules:
+ * https://developers.google.com/amp/cache/overview
+ * 
+ * Example:
+ * Input domain 'http://ampproject.org' can return 
+ * 'https://www-ampproject-org.cdn.ampproject.org/v/s/www.ampproject.org/?amp_js_v=0.1#origin=http%3A%2F%2Flocalhost%3A8000'
+ * 
+ * @param {string} domain The publisher domain
+ * @param {string} protocol The publisher domain
```

Comment:
s/domain/protocol/

Actually, it seems like it would make more sense to parse the protocol in this function since you also have the publisher URL.